app/assets/javascripts/components/common/ArticleViewer/containers/ArticleViewer.jsx

```diff
@@ -116,7 +124,13 @@ export class ArticleViewer extends React.Component {
     if (!this.props.users && !this.props.showArticleFinder) {
       this.props.fetchArticleDetails();
     } else if (!this.state.userIdsFetched && !this.props.showArticleFinder) {
-      this.fetchUserIds(this.props.users);
+      // if articleViewer is accessed through Students/Editors tab, a combination
+      // of both assignedUsers and users will be passed to the fetchUserIds function
+      // when show is clicked and the conditions have been satisfied. Otherwise
+      // whenever the articleViewer is accessed through any other tab, e.g Articles tab,
+      // the fetchUserIds function will be invoked with users because
+      // assignedUsers would be undefined.
+      this.fetchUserIds(union(this.props.assignedUsers || [], this.props.users));
```

Comment:
I think this is just about ready!

Since this is the key strategy for making the set of users consistent and it gets used in two different places, I think it would make sense to refactor this further. The arguments being passed to `fetchUserIds()` are just derived from props, so we don't actually need to pass them at all; constructing the array of users from the props could be done within the `fetchUserIds()` function, or a new function could be extracted which basically just does `this.fetchUserIds(union(this.props.assignedUsers || [], this.props.users));`. Either way, this would reduce duplication and give you only one place that requires documentation of what's going on.